### PR TITLE
DEPR: DTI/TDI/PI constructor arguments

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Comment by jbrockmendel** [comment]
The updated edit in the 0.15.0.rst is a shot in the dark, suggestions welcome on the appropriate way to handle it

File: doc/source/whatsnew/v0.15.0.rst [hunk 1]
```diff
@@ -311,8 +311,7 @@ Timezone handling improvements
 - ``tz_localize(None)`` for tz-aware ``Timestamp`` and ``DatetimeIndex`` now removes timezone holding local time,
   previously this resulted in ``Exception`` or ``TypeError`` (:issue:`7812`)
 
-  .. ipython:: python
-     :okwarning:
+  .. code-block:: ipython
```
Comment (jreback) [inline doc/source/whatsnew/v0.15.0.rst]: if you change to a code-block you have to show the evaluated code

File: pandas/core/indexes/period.py [hunk 2]
```diff
@@ -219,29 +216,9 @@ def __new__(
 
         if data is None and ordinal is None:
             # range-based.
-            data, freq2 = PeriodArray._generate_range(start, end, periods, freq, fields)
-            # PeriodArray._generate range does validate that fields is
+            data, freq2 = PeriodArray._generate_range(None, None, None, freq, fields)
```
Comment (jreback) [inline pandas/core/indexes/period.py]: does this actually work with None values?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/period.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/indexes/period.py]: Yes, PeriodArray._generate_range dispatches to either _get_ordinal_range or _range_from_fields

File: doc/source/whatsnew/v0.15.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v0.15.0.rst]: fixed+green

File: doc/source/whatsnew/v0.15.0.rst [hunk 3]
```diff
@@ -318,8 +318,8 @@ Timezone handling improvements
      ts
      ts.tz_localize(None)
 
-     didx = pd.DatetimeIndex(start='2014-08-01 09:00', freq='H',
-                             periods=10, tz='US/Eastern')
+     didx = pd.date_range(start='2014-08-01 09:00', freq='H',
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.0.rst]: can you remove the ":okwarning:" a few lines above?

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.15.0.rst]: okwarning removed + green

**Comment by jreback** [comment]
lgtm, can you add a deprecation removal note in whatsnew, ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
ping

**Pull request merged**